client/src/components/ProtectedRoute/ProtectedRoute.js

```diff
@@ -0,0 +1,29 @@
+import { Route, Redirect } from 'react-router-dom';
+
+//Protected route component to wrap pages that require users to be logged in beforehand.
+//Redirects to login page is user=false
+const ProtectedRoute = ({ component: Component, loggedInUser, ...rest }) => {
+  return (
+    <Route
+      {...rest}
+      render={(props) => {
+        if (loggedInUser) {
+          return <Component {...rest} {...props} />;
+        } else {
+          return (
+            <Redirect
+              to={{
+                pathname: '/login',
+                state: {
+                  from: props.location,
+                },
+              }}
+            />
```

Comment:
Can you alter this logic to handle the case where the API request has not come back yet? As is this would redirect a logged in user as well because this logic would fire before the app can determine if the user is logged in